snap/snapenv/snapenv_test.go

```diff
@@ -97,6 +97,17 @@ func (ts *HTestSuite) TestBasic(c *C) {
 	})
 }
 
+func (ts *HTestSuite) TestSaveDataEnvironment(c *C) {
+	dirs.SetRootDir(c.MkDir())
+	ts.AddCleanup(func() { dirs.SetRootDir("") })
+	c.Assert(os.MkdirAll(snap.CommonDataSaveDir(mockSnapInfo.InstanceName()), 0755), IsNil)
+
+	// The snap environment should now include SNAP_SAVE_DATA with the above path. It should
+	// only be added when the folder exists.
+	env := basicEnv(mockSnapInfo)
+	c.Assert(env["SNAP_SAVE_DATA"], Equals, snap.CommonDataSaveDir(mockSnapInfo.InstanceName()))
```

Comment:
one of the other tests should check that this is not present, no? or this one just before creating the dir

Reply:
Good point, I'll add an explicit test, fwiw the remaining tests in the file indirectly verify that the variable is not present as they verify the entire contents of the environment with DeepEquals, which would fail if the variable were to turn up.